### DISCUSSION TITLE
Disable github as datasource

### DISCUSSION THREAD

**Discussion by loewexy** [renovatebot/renovate#31311] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab with Renovate 38

### Please tell us more about your question or problem

I enabled Renovate on two repositories, one using Poetry and the other using npm. Since I am in a closed corporate environment Renovate does not have access to the internet. Instead we have internal mirrors of NPM and PyPi.

I set the URLs to our internal mirrors using packageRules, which match on the datasources npm and pypy respectively and set the registryUrls accordingly. This works and renovate can find updates there.

But renovate has for Poetry and npm still github as supported datasource and tries to retrieve data from there. Which fails because there is not internet access. Is there a possibility to disable specific datasources? Then I could just limit Poetry to use the internal pypi and npm to use the internal NPM mirror.

**Comment by rarkins** [comment, marked as answer]
Yes, use a package rule with matchDatasources and enabled=false

**Reply by loewexy** [reply]
This worked, however setting enabled=false in a packageRule was not considered valid according to the json schema for renovate. This was the reason I did not try this beforehand.
